libdevice/nativecpu_utils.cpp

```diff
@@ -36,16 +36,15 @@ using __nativecpu_state = native_cpu::state;
 DEVICE_EXTERN_C void __mux_work_group_barrier(uint32_t id, uint32_t scope,
                                               uint32_t semantics);
```

Comment:
```suggestion
DEVICE_EXTERN_C void __mux_work_group_barrier(int32_t id, int32_t scope,
                                              int32_t semantics);
```
And likewise below?

Reply:
done changes to __mux_work_group_barrier and __mux_mem_barrier in https://github.com/intel/llvm/pull/17438/commits/2b5b55d650b8f3c9685cf8f0fae0f32121084ef9

This would also require change in native_cpu backend, right?

Reply:
Thanks. This should not require any changes in the NativeCPU passes as they process this function at the LLVM IR level where both `uint32_t` and `int32_t` get mapped to the same `i32` type. NativeCPU changes look good to me, thanks.